sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -661,13 +666,19 @@ case class StructsToJson(
         (arr: Any) =>
           gen.write(arr.asInstanceOf[ArrayData])
           getAndReset()
+      case MapType(_: DataType, _: StructType, _: Boolean) =>
```

Comment:
Looks we can use this `MapType` directly for `mapType` in `gen.write(map.asInstanceOf[MapData], mapType)`.